### PR TITLE
Functions to append/prepend with printf formatting

### LINKED ISSUE #3: append/prepend sprintf

### PR DESCRIPTION
**Pull request by marcomorain**
Add buffer_appendf and bugger_prependf to allow the
user to append or prepend a string formatted with
printf-style syntax.

Fixes #3

### CODE REVIEW

**Comment by marcomorain** [comment]
Hi @stephenmathieson I've updated this – I've removed the prependf function (I don't use that in https://github.com/marcomorain/garcon) and implemented the appendf without vasnprintf.

File: buffer.c [hunk 1]
```diff
@@ -140,6 +141,35 @@ buffer_resize(buffer_t *self, size_t n) {
...
+    return -1;
+  }
+
+  // Next format the string into the space that we have made room for.
+  char* dst = self->data + initial_len;
+  const int bytes_formatted = vsnprintf(dst, 1+space_required, format, ap);
+
+  printf("Result '%s'\n", self->data);
+  va_end(ap);
+  return (bytes_formatted < 0) ? -1 : 0;
```
Comment (marcomorain) [inline buffer.c]: any suggestions on how to make this line nicer? It feels a bit janky.

Comment (stephenmathieson) [inline buffer.c]: well the parentheses are unnecessary, but they dont hurt anything either.. i generally format ternary statements like:

```c
return a > b
  ? c
  : d
```

but i'm not sure it makes anything "cleaner".

---

i think the `return` is fine

**Comment by stephenmathieson** [comment]
thanks @marcomorain!

**Pull request merged**